### PR TITLE
Add a test to verify goroutine leak in compression pools.

### PR DESCRIPTION
**Pull request by cyriltovena**
I thought zstd has a leak for the writer, but it did not.

This also fixes a request from @kavirajk when I added the finalizer on the zstd reader.

Signed-off-by: Cyril Tovena <cyril.tovena@gmail.com>

### CODE REVIEW

File: pkg/chunkenc/pool_test.go [hunk 1]
```diff
@@ -50,11 +50,10 @@ func TestPool(t *testing.T) {
...
 	wg.Wait()
 
-	runtime.GC()
-	time.Sleep(20 * time.Millisecond)
-	runtime.GC()
-
-	if !assert.LessOrEqual(t, runtime.NumGoroutine(), 100) {
+	if !assert.Eventually(t, func() bool {
+		runtime.GC()
+		return runtime.NumGoroutine() <= 50
```
Comment (kavirajk) [inline pkg/chunkenc/pool_test.go:55]: Curious, why it's 50?

Comment (cyriltovena) [reply pkg/chunkenc/pool_test.go:55]: It's below 100 and account for other test running in parallel.

Comment (cyriltovena) [reply pkg/chunkenc/pool_test.go:55]: If there was a leak it would be higher than 200 or more.

Comment (kavirajk) [reply pkg/chunkenc/pool_test.go:55]: Hmm. thanks for explaining! Doesn't this also mean, this number will become flaky as we keep adding more and more parallel tests somewhere else?

(not a blocker for the PR)

**Review by kavirajk — APPROVED** [review]
LGTM!

File: pkg/chunkenc/pool_test.go (hunk 1 above)
Comment (cyriltovena) [reply pkg/chunkenc/pool_test.go:55]: Yeah That's why I hesitate to get it in the first place, but then I thought there was a leak again. So I wrote a test. Normally go should run that many test in parallel though, it's also only possible for the current package.

**Pull request merged**